src/main.rs

```diff
@@ -63,6 +62,43 @@ impl IdentityMappedAddr {
     }
 }
 
+// Symbols defined in `linker.ld`
+extern {
+    static _kib_kernel_size: usize;
+    static _memory_map: usize;
+    static mmap_ent: usize;
+    static __page_table_start: usize;
+    static __page_table_end: usize;
+    static __bootloader_end: usize;
+    static __bootloader_start: usize;
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn stage_4() -> ! {
+    asm!("mov bx, 0x0" :::: "intel");
+    asm!("mov ss, bx" :::: "intel"); // Set stack segment
+
+    let kernel_start = 0x400000;
+    let kernel_size = _kib_kernel_size as u64;
+    let memory_map_addr = &_memory_map as *const _ as u64;
+    let memory_map_entry_count = 6;
```

Comment:
This is just hard-coded: from my PR description you can see I don't know how this value always resolves to `6`.